rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/edition/EditNodeSourceWindow.java

```diff
@@ -63,6 +63,7 @@ protected EditNodeSourceWindow(RMController controller, String nodeSourceName, S
     private void setAttributesAndBuildForm(String nodeSourceName) {
         this.nodeSourceName = nodeSourceName;
         buildForm();
+        buildForm();
```

Comment:
Maybe it's a stupid question, but why do we have to call `buildForm()` two times?

Reply:
It is not stupid! I dont know either why. Before for editing dynamic parameters, we were calling it twice (once in EditDynamicParametersWindow constructor, and once in its parent EditNodeSourceWindow). But to edit undeployed NS, we were calling EditNodeSourceWindow which calls buildForm only once. It does not work when called once. So now, for both editing dynamic parameters and editing undeployed NS we will call it twice.

Reply:
That's really weird xD